Gruntfile.coffee

```diff
@@ -1,108 +1,286 @@
 TaskManager = require './tools/taskmanager'
-Rule = require './tools/common-grunt-rules'
+
+#-------------------------------------------------------------------------
+# The top level tasks. These are the highest level grunt-tasks defined in terms
+# of specific grunt rules below and given to grunt.initConfig
+taskManager = new TaskManager.Manager();
+
+# Setup makes sure the needed typescript definition files are in the right place
+# so that typescript compilation can find them using require and defines
+# references in the code.
+taskManager.add 'setup', [
+  'tsd:dev'
+]
+
+# This rebuilds the tools directory. It should not often be needed.
+taskManager.add 'tools', [
+  'ts:dev'
+  'copy:tools'
+]
+
+# Makes the base development build, excludes sample apps.
+taskManager.add 'base-dev', [
+  'copy:dev'
+  'ts:dev'
+  'browserify:loggingProvider'
+]
+
+# Makes the development build, includes sample apps.
+taskManager.add 'dev', [
```

Comment:
this is the same task as sample, if I'm reading definitions correctly. consider condensing to 1 name?